crates/services/shared-sequencer/src/service.rs

```diff
@@ -0,0 +1,262 @@
+            account_metadata: None,
+            prev_order: None,
+            blobs,
+        })
+    }
+}
+
+impl<S> Task<S>
+where
+    S: Signer,
+{
+    async fn blobs(&mut self) -> anyhow::Result<Option<SSBlobs>> {
+        let ss = self
+            .shared_sequencer_client
+            .as_ref()
+            .expect("Shared sequencer client is not set; qed");
+
+        if self.account_metadata.is_none() {
+            // If the account is not funded, this code will fail
+            // because we can't sign the transaction without account metadata.
+            let account_metadata = ss.get_account_meta(self.signer.as_ref()).await;
+
+            match account_metadata {
+                Ok(account_metadata) => {
+                    self.account_metadata = Some(account_metadata);
+                }
+                Err(err) => {
+                    tokio::time::sleep(Duration::from_secs(1)).await;
+                    return Err(err);
+                }
+            }
+        }
+
+        if self.prev_order.is_none() {
+            self.prev_order = ss.get_topic().await?.map(|f| f.order);
+        }
+
+        tokio::time::sleep(self.config.block_posting_frequency).await;
+
+        let blobs = {
+            let mut lock = self.blobs.lock().await;
+            core::mem::take(&mut *lock)
+        };
+
+        if blobs.is_empty() {
+            tokio::time::sleep(Duration::from_secs(1)).await;
+            Ok(None)
+        } else {
+            Ok(Some(blobs))
+        }
+    }
+}
+
+const CONTINUE: bool = true;
+const STOP: bool = false;
+
+#[async_trait]
+impl<S> RunnableTask for Task<S>
+where
+    S: Signer + 'static,
+{
+    async fn run(&mut self, watcher: &mut StateWatcher) -> anyhow::Result<bool> {
+        if !self.config.enabled {
+            let _ = watcher.while_started().await;
+            return Ok(STOP);
+        }
+
+        tokio::select! {
+            biased;
+            _ = watcher.while_started() => {
+                Ok(STOP)
+            },
+
+            blobs = self.blobs() => {
+                let blobs = blobs?;
+
+                if let Some(blobs) = blobs {
+                    let mut account = self.account_metadata.take().expect("Account metadata is not set; qed");
+                    let next_order = if let Some(prev_order) = self.prev_order {
+                        prev_order.wrapping_add(1)
+                    } else {
+                        0
+                    };
+
+                    let ss =  self.shared_sequencer_client
+                        .as_ref().expect("Shared sequencer client is not set; qed");
+                    let blobs_bytes = postcard::to_allocvec(&blobs).expect("Failed to serialize SSBlob");
+                    let result = ss.send(self.signer.as_ref(), account, next_order, blobs_bytes).await;
+
+                    match result {
+                        Ok(_) => {
+                            tracing::info!("Posted block to shared sequencer {blobs:?}");
```

Comment:
Do we want to log the full bytes in the release at an info level ?

Reply:
Yes, the log is very small and only posted once per 12 seconds.